### PR TITLE
Nightly build scripts

### PR DESCRIPTION
**Pull request by ultimaterex**
This provides an entry point for web and ios builds.
This commit also has a few improvements and fixes for the sync branches script. This should now correctly update the nightly build even when fast forwarding with the head is possible. this also amends the commit in the nightly branch with the correct nightly version. 
The android version has been setup to allow for passing in the nightly build version via dart define and will now also increment the build version.

### CODE REVIEW

File: .github/workflows/nightly_build_ios.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      # Prints version to the console
+      - name: Flutter Version
+        run: flutter --version
+
+      # Creates generated files
+      - name: Install flutter dependencies
+        run: make gen
+
+      - name: Build iOS app
+        run: flutter build ios --release --no-codesign
```
Comment (KalervoHyyppa) [inline .github/workflows/nightly_build_ios.yml:43]: this will need to be flutter build ipa

File: .github/workflows/nightly_build_web.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    branches:
+      - nightly
+  workflow_dispatch:
+
+env:
+  FLUTTER_CHANNEL: "stable"
+
+jobs:
+  build_chrome:
+    name: Build Flutter (Android)
```
Comment (KalervoHyyppa) [inline .github/workflows/nightly_build_web.yml:17]: Change to "Build Flutter (Web)"

File: .github/workflows/nightly_build_web.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    runs-on: macOS-latest
```
Comment (KalervoHyyppa) [inline .github/workflows/nightly_build_web.yml:18]: Does it matter if we use macOs? I believe using ubuntu might be a bit faster

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
Could you also change the base to `main`

File: .github/workflows/nightly_build_ios.yml (hunk 1 above)
Comment (ultimaterex) [reply .github/workflows/nightly_build_ios.yml:43]: Added also, added the snippet for dart define and the build number

File: .github/workflows/nightly_build_web.yml (hunk 3 above)
Comment (ultimaterex) [reply .github/workflows/nightly_build_web.yml:18]: In true apple way we can't compile iOS/MACOS apps outside of Apple's ecosystem, so unfortunately macOS is the only thing we've got

Comment (KalervoHyyppa) [reply .github/workflows/nightly_build_web.yml:18]: But isn't this for web? I don't think we need macOs for web

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
Check out nightly_build_web

**Review by KalervoHyyppa — APPROVED** [review]
LGTM

**Pull request merged**